File: client/src/app/classes/message-game-replay-command/message-game-replay-command.ts
import { GameReplayCommand } from '@app/interfaces/game-replay-command';
import { ClassicModeService } from '@app/services/classic-mode/classic-mode.service';
// import { ChatMessage } from '@common/model/message';
import { ReplayEventDto } from '@common/model/events/replay.events';

export class MessageGameReplayCommand implements GameReplayCommand {
    time: number;
    // private classicModeService: ClassicModeService;
    // private replayEventDto: ReplayEventDto;

    constructor(classicModeService: ClassicModeService, replayEventDto: ReplayEventDto) {
        this.time = replayEventDto.time;
        // this.classicModeService = classicModeService;
        // this.replayEventDto = replayEventDto;
    }

    action() {
        // FIX ME this.classicModeService.receiveMessage(this.replayEventDto.body as ChatMessage);
    }
}

File: client_leger/lib/classes/game-replay-commands/message_game_replay_command.dart
import 'package:client_leger/classes/game-replay-commands/game_replay_command.dart';
import 'package:client_leger/interfaces/dto/replay_event.dto.dart';

class MessageGameReplayCommand extends GameReplayCommand {
  final MessageReplayDto eventDto;

  MessageGameReplayCommand({required this.eventDto, required super.time});

  @override
  void action() {
    print('MessageGameReplayCommand action');
  }
}
